control-plane-operator/controllers/hostedcontrolplane/olm/catalogs.go

```diff
@@ -148,28 +148,31 @@ func GetCatalogImages(ctx context.Context, hcp hyperv1.HostedControlPlane, pullS
 	//check catalogs of last 4 supported version in case new version is not available
 	supportedVersions := 4
 	imageRegistry := ""
-	for i := 0; i < supportedVersions; i++ {
-		for _, registry := range registries {
-			testImage := fmt.Sprintf("%s/certified-operator-index:v%d.%d", registry, version.Major, version.Minor)
+	if hcp.Spec.OLMCatalogPlacement == hyperv1.ManagementOLMCatalogPlacement {
```

Comment:
As far as I can tell, I think we need to set `imageRegistry` to `registry.redhat.io/redhat` when using guest placement else it will be empty.  Did you test this?  Also, I think it would be good to add a unit test for this.